Automattic/jetpack
# Jetpack: fix bug wen getting the video preview of the VPBlock

This PR removes the part of the code where the VideoPress URL is stored in the local state. Instead of it, it just updates the `guid` and `src` block attributes, guaranteeing that the VideoPress URL will update too according its parameters (controls, autoplay, etc).
In terms of the UI, it fixes

* Getting the preview just after the video is uploaded
* Refresh the preview when changing the video params (controls, autoplay, etc)
* It renders the preview using the cache HTML when it exists (initial rendering)

Fixes #

#### Changes proposed in this Pull Request:

* Jetpack: fix bug wen getting the video preview of the VPBlock

#### Other information:



#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Go to block editor
* Add a new VideoPress block
* Uploads a video
* Confirm that, with these changes, the preview is generated on the fly
* Confirm the preview is re-generated when changing video attributes
* Confirm that, after a hard refresh, the preview takes less time to get the initial preview

The following video shows how it behaves

https://user-images.githubusercontent.com/77539/176910649-330cfa0b-9b6b-435d-ac75-938790ab416e.mov




---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202528984221997
